### PR TITLE
API: fixed callback of PrecomputedInterpolator injection

### PR DESCRIPTION
**Pull request by ccuetom**
Although the interpolation using PrecomputedInterpolator was working fine, the injection tries to access the `gridpoints` and `interpolation_coeffs` in `self` instead of in `self.obj`.

Also, when injecting with the PrecomputedInterpolator, the preparation of the arguments for the operator calls the `self._parent._arg_defaults` for the gridpoints SubFunction, which does not exist. By changing the parent of the SubFunction to `self.obj` this is solved.

### CODE REVIEW

File: tests/test_interpolation.py [hunk 1]
```diff
@@ -169,6 +169,42 @@ def test_precomputed_interpolation_time():
...
+def test_precomputed_injection():
+    """Test injection with PrecomputedSparseFunction which accepts
+       precomputed values for interpolation coefficients
+    """
+    shape = (11, 11)
+    coords = [(.05, .95), (.45, .45)]
+    origin = (0, 0)
+    result = 0.25
+
+    r = 2  # Constant for linear interpolation
```
Comment (georgebisbas) [inline tests/test_interpolation.py:181]: nitpicking:
Comment should go before variable?

File: tests/test_interpolation.py [hunk 2]
```diff
@@ -169,6 +169,42 @@ def test_precomputed_interpolation_time():
...
+    #  because we interpolate across 2 neighbouring points in each dimension
+
+    m = unit_box(shape=shape)
+    m.data[:] = 0.
+
+    r = 2  # Constant for linear interpolation
```
Comment (georgebisbas) [inline tests/test_interpolation.py:187]: Duplicate?

**Review by georgebisbas — CHANGES_REQUESTED** [review]

File: tests/test_interpolation.py (hunk 1 above)
Comment (ccuetom) [reply tests/test_interpolation.py]: I have now moved the comments before the variable

File: tests/test_interpolation.py (hunk 2 above)
Comment (ccuetom) [reply tests/test_interpolation.py]: I have removed the duplicate variable

**Comment by FabioLuporini** [comment]
Thanks a lot, merging!

**Pull request merged**